src/swarm-engine/Swarm/Game/Step/Const.hs

```diff
@@ -1542,7 +1542,7 @@ execConst runChildProg c vs s k = do
         -- already has.
         missingChildInv = reqInv `E.difference` childInventory
 
-    if creative
+    if privileged
```

Comment:
Should all privileged robots always be able to create devices + entities out of thin air like this?

Right now, if robot A is a system robot and tries to create a child robot B which (say) executes `move`, it won't be able to if it doesn't have `treads` in its inventory.  This change represents one way to fix this, which is to simply allow robot A to conjure `treads` out of thin air.  However, in a way this seems unnecessary, since robot B will also be a system robot and hence will not actually need the `treads`.

Reply:
> Should all privileged robots always be able to create devices + entities out of thin air like this?

I think so.

I suppose the alternative would be to just allow the `build` command to succeed, without equipping anything?  I'd be fine with that as well.  One goal of this change for me, is to minimize "forking" of behavior between "creative" mode and "system" robots.  I wouldn't want it to be that in "creative" mode, all of the requisite devices are equipped, but for "system" robots (not in creative mode), we bypass the `build` command's restrictions by a different code path.

Reply:
> I suppose the alternative would be to just allow the build command to succeed, without equipping anything?

Right, that's the alternative I had in mind.  In a sense this alternative feels more "correct" from an in-game logic point of view.  System robots can just do whatever they want; they don't need to equip devices on other system robots they build, since those other robots can also just do whatever they want.  However:

> One goal of this change for me, is to minimize "forking" of behavior between "creative" mode and "system" robots.

That's a very good point.  Let's keep things simple for now, and only make things more complex if really necessary at some point in the future.